lib/instrumentation/http-shared.js

```diff
@@ -260,3 +257,40 @@ function isAWSSigned (opts) {
   const auth = opts.headers && (opts.headers.Authorization || opts.headers.authorization)
   return typeof auth === 'string' ? auth.startsWith('AWS4-') : false
 }
+
+// Creates a sanitized URL suitable for the span's HTTP context
+//
+// This function reconstructs a URL using the request object's properties
+// where it can, and falling back to the options where it can not.
+//
+// NOTE: The options argument may not be the same options that are passed
+// to http.request if the caller uses the the http.request(url,options,...)
+// method signature. The agent normalizes the url and options into a single
+// options array. This function expects those pre-normalized options.
+//
+// @param {ClientRequest} req
+// @param {object} options
+// @return string
+function getUrlFromRequestAndOptions (req, options) {
+  if (!req) {
+    return false
```

Comment:
Returning false here results in `"url":false` in the span. E.g.:

```
{"span":{"name":"GET httpbin.org/Get","type":"external",..."duration":190.626,"context":{"http":{"method":"GET","status_code":404,"url":false}}...
```

Reply:
switched this up from false to undefined to ensure its dropped from the serialized span.